### PR TITLE
mcap 0.0.6 (new formula)

### PR DESCRIPTION
**Pull request by jtbandes**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This PR adds a new formula for the [MCAP](https://github.com/foxglove/mcap) project's CLI tool. MCAP is a modular, performant, and serialization-agnostic container file format for pub/sub messages, primarily intended for use in robotics applications.

MCAP also has libraries available for several languages (Go, C++, Python, TypeScript, Swift) but this formula is for the CLI tool only.

### CODE REVIEW

**Comment by jtbandes** [comment]
I see the test is failing on linux with error message:
`Failed to create reader: not an mcap file`

This would seem to indicate the test file is corrupt, however the checksum validation seems to have passed. I also just tried installing linuxbrew in an ubuntu 20.04 VM, pulling this branch, and was able to successfully run:

```bash
$ brew test mcap --verbose
==> Testing mcap
cp -p /home/parallels/.cache/Homebrew/downloads/1d8e3f4d5a767e6ff52e81aad50d3ffa767e859ef219fbd7524e8ffb540cc19c--OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap /tmp/mcap--testdata-20220408-211631-47hzj5/OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap
==> /home/parallels/.linuxbrew/Cellar/mcap/0.0.5/bin/mcap version
==> /home/parallels/.linuxbrew/Cellar/mcap/0.0.5/bin/mcap cat OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap
==> /home/parallels/.linuxbrew/Cellar/mcap/0.0.5/bin/mcap info OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap
```

Any other ideas for why this would be failing on linux? Note that the test file being downloaded is [stored with Git LFS](https://github.com/foxglove/mcap/blob/releases/mcap-cli/v0.0.5/tests/conformance/data/OneMessage/OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap) (however the `media.githubusercontent.com` url does seem to correctly return the actual file contents, not the LFS pointer file).

**Comment by jtbandes** [comment]
Hi @Bo98, thanks for helping with the CI runs. Now that the tests are passing, is there anything else I need to do to make this PR ready for review?

File: Formula/mcap.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  desc "Serialization-agnostic container file format for pub/sub messages"
+  homepage "https://mcap.dev"
+  url "https://github.com/foxglove/mcap/archive/refs/tags/releases/mcap-cli/v0.0.6.tar.gz"
+  sha256 "f2adc1a2c1f81d4c5f5b684829013dcdf28f24f15ce34e4a72139ee01355d93a"
+  license "Apache-2.0"
+  head "https://github.com/foxglove/mcap.git", branch: "main"
+
+  depends_on "go" => :build
+
+  resource "testdata" do
```
Comment (SMillerDev) [inline Formula/mcap.rb:11]:
```suggestion
  resource "homebrew-testdata" do
```
This allows the update script to ignore it

File: Formula/mcap.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  def install
+    cd "go/cli/mcap" do
+      system "make", "build", "VERSION=v#{version}"
+      bin.install "bin/mcap"
+    end
+  end
+
+  test do
+    resource("testdata").stage do
+      assert_equal "v#{version}", pipe_output("#{bin}/mcap version").strip
```
Comment (SMillerDev) [inline Formula/mcap.rb:25]: This can be outside the resource

File: Formula/mcap.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      assert_equal "2 example [Example] [1 2 3]",
+        shell_output("cat OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap | #{bin}/mcap cat").strip
```
Comment (SMillerDev) [inline Formula/mcap.rb:28]:
```suggestion
        pipe_output(File.read "OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap", "#{bin}/mcap cat").strip
```

**Comment by SMillerDev** [comment]
Hi, this seems like a nice software, but not notable enough for us to package in Homebrew for now. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Pull request closed without merging**